tools/eksDistroBuildToolingOpsTools/cmd/eksDistroOpsProwPlugin/server.go

```diff
@@ -0,0 +1,269 @@
+		if err := json.Unmarshal(payload, &ie); err != nil {
+			return err
+		}
+		go func() {
+			if err := s.handleIssue(l, ie); err != nil {
+				s.log.WithError(err).WithFields(l.Data).Info("Issue creation failed.")
+			}
+		}()
+	default:
+		logrus.Debugf("skipping event of type %q", eventType)
+	}
+	return nil
+}
+
+func (s *Server) handleIssue(l *logrus.Entry, ie github.IssueEvent) error {
+	// Only consider newly opened issues and not PRs
+	if ie.Action != github.IssueActionOpened && !ie.Issue.IsPullRequest() {
+		return nil
+	}
+
+	org := ie.Repo.Owner.Login
+	repo := ie.Repo.Name
+	num := ie.Issue.Number
+	auth := ie.Sender.Name
+	title := ie.Issue.Title
+	body := ie.Issue.Body
+
+	// Do not create a new logger, its fields are re-used by the caller in case of errors
+	*l = *l.WithFields(logrus.Fields{
+		github.OrgLogField:  org,
+		github.RepoLogField: repo,
+		github.PrLogField:   num,
+	})
+
+	if auth != "eks-distro-bot" || !s.allowAll {
+		ok, err := s.ghc.IsMember(org, auth)
+		if err != nil {
+			return err
+		}
+		if !ok {
+			resp := fmt.Sprintf("only [%s](https://github.com/orgs/%s/people) org members may request may trigger automated issues. You can still create the issue manually.", org, org)
+			l.Info(resp)
+			return s.ghc.CreateComment(org, repo, num, resp)
+		}
+	}
+
+	//Currently handling Golang Patch Releases and Golang Minor Releases
+	var golangPatchReleaseRe = regexp.MustCompile(`(?m)^(?:Golang Patch Release:)\s+(.+)$`)
+	//var golangMinorReleaseRe = regexp.MustCompile(`(?m)^(?:Golang Minor Release:)\s+(.+)$`)
+
+	golangPatchMatches := golangPatchReleaseRe.FindAllStringSubmatch(ie.Issue.Title, -1)
+	if len(golangPatchMatches) != 0 {
+		if err := s.handle(l, ie.Issue.User.Login, &ie.Issue, org, repo, title, body, num); err != nil {
+			return fmt.Errorf("handle GolangPatchrelease: %w", err)
+		}
+	}
+	//TODO: add golangMinorMatches := golangMinorReleaseRe.FindAllStringSubmatch(ie.Issue.Title, -1)
+
+	return nil
+}
+
+func (s *Server) handle(logger *logrus.Entry, requestor string, issue *github.Issue, org, repo, title, body string, num int) error {
+	var lock *sync.Mutex
+	func() {
+		s.mapLock.Lock()
+		defer s.mapLock.Unlock()
+		if _, ok := s.lockMap[upstreamPickRequest{org, repo, num}]; !ok {
+			if s.lockMap == nil {
+				s.lockMap = map[upstreamPickRequest]*sync.Mutex{}
+			}
+			s.lockMap[upstreamPickRequest{org, repo, num}] = &sync.Mutex{}
+		}
+		lock = s.lockMap[upstreamPickRequest{org, repo, num}]
+	}()
+	lock.Lock()
+	defer lock.Unlock()
+
+	if err := s.HandleGolangPatchRelease(logger, issue); err != nil {
+		return fmt.Errorf("failed to handle Golang Patch Release: %w", err)
+	}
+
+	return nil
+}
+
+// Created based off plugins.FormatICResponse
+func FormatIEResponse(ie github.IssueEvent, s string) string {
+	return plugins.FormatResponseRaw(ie.Issue.Title, ie.Issue.HTMLURL, ie.Sender.Login, s)
+}
+
+func (s *Server) HandleGolangPatchRelease(l *logrus.Entry, upIss *github.Issue) error {
```

Comment:
Golint comments: exported method Server.HandleGolangPatchRelease should have comment or be unexported. [More info](https://golang.org/wiki/CodeReviewComments#doc-comments). <!-- golint -->